arviz/tests/base_tests/test_utils.py

```diff
@@ -64,6 +57,36 @@ def test_var_names_key_error(data):
         _var_names(("theta", "tau", "bad_var_name"), data)
 
 
+@pytest.mark.parametrize(
+    "var_names_expected",
+    [
+        (["alpha", "beta"], ["alpha", "beta1", "beta2"], "like"),
+        (["~beta"], ["alpha", "p1", "p2", "phi"], "like"),
+        (["p"], ["alpha", "p1", "p2", "phi"], "like"),
+        (["~p"], ["beta1", "beta2"], "like"),
+        (["^bet"], ["beta1", "beta2"], "regex"),
+        (["^p"], ["p1", "p2", "phi"], "regex"),
+        (["~^p"], ["alpha", "beta1", "beta2"], "regex"),
+        (["p[0-9]+"], ["p1", "p2"], "regex"),
+        (["~p[0-9]+"], ["alpha", "beta1", "beta2", "phi"], "regex"),
+    ],
+)
+def test_var_names_filter(var_names_expected):
+    """Test var_name filter with partial naming or regular expressions."""
+    data = from_dict(
+        posterior={
+            "alpha": np.random.randn(10),
+            "beta1": np.random.randn(10),
+            "beta2": np.random.randn(10),
+            "p1": np.random.randn(10),
+            "p2": np.random.randn(10),
+            "phi": np.random.randn(10),
+        }
+    ).posterior
```

Comment:
I think this could use `dict_to_dataset` to avoid doing the `.posterior` and get the dataset straight away.